examples/stress_tests/bevymark.rs

```diff
@@ -221,10 +222,10 @@ fn setup(
         quad: meshes
             .add(Rectangle::from_size(Vec2::splat(BIRD_TEXTURE_SIZE as f32)))
             .into(),
-        color_rng: StdRng::seed_from_u64(42),
-        material_rng: StdRng::seed_from_u64(42),
-        velocity_rng: StdRng::seed_from_u64(42),
-        transform_rng: StdRng::seed_from_u64(42),
+        color_rng: ChaCha8Rng::seed_from_u64(42),
```

Comment:
`ChaCha8Rng` implements `Clone` I think, so if we want, we could just clone a single instance over and over if they are all using the same seed though. Buuuut, it seems like a bit of a code smell to reuse the same seed for multiple RNG instances, even if an example. If we want things to be deterministic with generated seeds, we could make use of a single `ChaCha8Rng::seed_from_u64(42)`, and then everything else gets `ChaCha8Rng::from_rng(&mut rng).unwrap()`.

Reply:
The documentation says the use of `unwrap` is discouraged, do you think it's fine here or shall I try to use a `match` or something? (I'm very new to Rust, so apologies, I'll have more questions than usual!)

Reply:
> `ChaCha8Rng` implements `Clone` I think, so if we want, we could just clone a single instance over and over if they are all using the same seed though. Buuuut, it seems like a bit of a code smell to reuse the same seed for multiple RNG instances, even if an example. If we want things to be deterministic with generated seeds, we could make use of a single `ChaCha8Rng::seed_from_u64(42)`, and then everything else gets `ChaCha8Rng::from_rng(&mut rng).unwrap()`.

I disagree with cloning in this case. I think it will make reading the code more difficult, since the variable will need to be created beforehand. As for changing the seeds, it may be good practice in an actual game but changing the seeds now will mess up [the screenshot tests](https://thebevyflock.github.io/bevy-example-runner/). I don't think it's worth the change.

Reply:
> The documentation says the use of `unwrap` is discouraged, do you think it's fine here or shall I try to use a `match` or something? (I'm very new to Rust, so apologies, I'll have more questions than usual!)

For OS/Hardware sources which can fail, then yes, you shouldn't `unwrap()`. But for deterministic pseudorandom sources where you are in full control of the seeding (such as with `ChaCha8Rng`), then usage of the Rng source will never fail. Not all entropy sources are fallible, only when they interface hardware/OS APIs can failures happen. If a PRNG algorithm can fail, then it is not deterministic in output.

Reply:
> I disagree with cloning in this case. I think it will make reading the code more difficult, since the variable will need to be created beforehand. As for changing the seeds, it may be good practice in an actual game but changing the seeds now will mess up [the screenshot tests](https://thebevyflock.github.io/bevy-example-runner/). I don't think it's worth the change.

Then perhaps we need some comments here to explain that a bit more clearly. But yeah, if we have already a bunch of screenshot test suites in place, we shouldn't change that unless absolutely necessary.